go/vt/vtgate/planbuilder/operators/insert.go

```diff
@@ -0,0 +1,123 @@
+/*
+Copyright 2023 The Vitess Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package operators
+
+import (
+	"vitess.io/vitess/go/vt/sqlparser"
+	"vitess.io/vitess/go/vt/vtgate/evalengine"
+	"vitess.io/vitess/go/vt/vtgate/planbuilder/operators/ops"
+	"vitess.io/vitess/go/vt/vtgate/vindexes"
+)
+
+type Insert struct {
+	VTable *vindexes.Table
+	AST    *sqlparser.Insert
+
+	AutoIncrement *Generate
+	Ignore        bool
+
+	// ColVindexes are the vindexes that will use the VindexValues
+	ColVindexes []*vindexes.ColumnVindex
+
+	// VindexValues specifies values for all the vindex columns.
+	// This is a three-dimensional data structure:
+	// Insert.Values[i] represents the values to be inserted for the i'th colvindex (i < len(Insert.Table.ColumnVindexes))
+	// Insert.Values[i].Values[j] represents values for the j'th column of the given colVindex (j < len(colVindex[i].Columns)
+	// Insert.Values[i].Values[j].Values[k] represents the value pulled from row k for that column: (k < len(ins.rows))
+	VindexValues [][][]evalengine.Expr
```

Comment:
would it be possible to hide the complexity of this data structure behind a few methods? My head is barely able to handle two dimensional arrays